generator-foodq/generators/app2/index.js

```diff
@@ -3,32 +3,63 @@ var Generator = require('yeoman-generator');
 module.exports = class extends Generator {
   constructor(args, opts) {
     super(args, opts);
-
-    this.getPrompts = function() {
-      console.log('in getPrompts()');
-      return [{name:"Prompt 1 app2"},{name: "Prompt 2 app2"}];
-    }
+    this.prompts = opts.prompts;
   }
 
   async prompting() {
+    // isTakeaway question appears twice in CLI: https://github.com/yeoman/generator/issues/1100
     let prompts = [
       {
         type: "confirm",
         name: "isTakeaway",
         message: "Do you want to take away?"
       },
       {
-        when: (response)=>{
-          return response.isTakeaway;
-        },
-        type: "input",
-        name: "address",
-        message: "Your Address"
+        type: "list",
+        name: "deliveryMethod",
+        message: "Delivery method",
+        choices: [
+          "Car",
+          "Drone"
+        ],
+      when: async (answers) => {
+          let indexOfAddress = -1;
+        for await (let [i, prompt] of this.prompts.items.entries()) {
+          if (prompt && prompt.name === "Address") {
+            indexOfAddress = i;
+          }
+        }
+
+        if (answers.isTakeaway) {
+          // add address prompt if doesn't exist
+          if (indexOfAddress === -1) {
+            this.prompts.splice(4, 0, { name: "Address", description: "Address Description" });
+          }
+        } else {
+          // remove address prompt if exists
+          if (indexOfAddress > -1) {
+            this.prompts.splice(4, 1);
```

Comment:
I am expecting the the sub generator woldn't handle any promts of root generator.
The sub generator could be a 'root' generator by himself and can declare setPromptsCallback in it's constructor.